retrybp/retry.go

```diff
@@ -0,0 +1,60 @@
+package retrybp
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	retry "github.com/avast/retry-go"
+
+	"github.com/reddit/baseplate.go/batcherror"
+)
+
+func init() {
+	retry.DefaultAttempts = uint(1)
+	retry.DefaultDelay = 5 * time.Millisecond
+	retry.DefaultMaxJitter = 25 * time.Millisecond
+}
```

Comment:
🔕 I have a few concerns with this approach, but I don't know if there's a better way to do it:

1. Those values might still be too high for some latency sensitive services, so I could see some devs still want to tweak them in their code, but as long as they don't use `init` (the order of `init` functions from different packages are not guaranteed) to do that that shouldn't be a big problem.
2. These values here are more specific for this version of upstream we are using here, but service could upgrade them to a newer version, which might have changes to the default values, and _might_ change the meaning of some of those default values, and cause problems. With go's no breaking change policy those shouldn't happen, but there's no guarantee of it.

One alternative is to have an exported function, something like `OverrideDefaultValues`, so service code could do the overriding explicitly. But I do see one disadvantage of that approach, is they might forget to do that and end up with the higher defaults (which using `init` avoids).

Reply:
I wonder if it's worth attaching a state to those retry functions to ensure the values set are the ones that the applier wants to have.